### PR TITLE
Small patches for routing daemon interactions

### PR DESCRIPTION
**Pull request by cjdoucette**

### CODE REVIEW

File: cps/kni.c [hunk 1]
```diff
@@ -1072,7 +1073,17 @@ rd_getroute(const struct nlmsghdr *req, struct cps_config *cps_conf, int *err)
 	family = ((struct rtgenmsg *)mnl_nlmsg_get_payload(req))->rtgen_family;
 
 	/* We don't support AF_UNSPEC to dump both tables at once. */
-	if (family != AF_INET && family != AF_INET6) {
```
Comment (AltraMayor) [inline cps/kni.c]: If the kernel supports `AF_UNSPEC` to dump both tables at once, please another patch to do so.

Comment (AltraMayor) [reply cps/kni.c]: Sorry. Ignore this comment.

**Comment by AltraMayor** [comment]
Can you verify that all the RTNetlink interface in Gatekeeper is checking if IPv4 and IPv6 is enabled where needed?

**Comment by cjdoucette** [comment]
Just rebased (and added a couple of new patches), but the patches already approved haven't been touched.

File: cps/kni.c [hunk 2]
```diff
@@ -734,7 +742,8 @@ attr_get(struct route_update *update, int family, struct nlattr *tb[])
 
 	update->valid = dst_present &&
 		(update->type == RTM_DELROUTE ||
-		(update->type == RTM_NEWROUTE && gw_present && oif_present));
+		(update->type == RTM_NEWROUTE && gw_present && oif_present) ||
+		(update->rt_type == RTN_BLACKHOLE));
```
[the remark this answers is not in the document]
Comment (cjdoucette) [reply cps/kni.c]: We could be deleting a blackhole entry as well, right? Perhaps the check should be it’s either a newroute or delroute AND a blackhole.

[the remark this answers is not in the document]
Comment (AltraMayor) [reply cps/kni.c]: To delete a route, the test only requires `dst_present`, which is enough for blackholes as well, so the enhanced test is only for `RTM_NEWROUTE`.

File: cps/kni.c [hunk 3]
```diff
@@ -497,6 +500,11 @@ new_route(struct route_update *update, const struct cps_config *cps_conf)
 		sizeof(prefix_info.addr.ip));
 	prefix_info.len = update->prefix_len;
 
+	if (update->rt_type == RTN_BLACKHOLE) {
```
[the remark this answers is not in the document]
Comment (AltraMayor) [reply cps/kni.c]: Not to mention that the gateway field won't be available, so it's likely all zeros.

File: cps/kni.c (hunk 2 above)
[the remark this answers is not in the document]
Comment (cjdoucette) [reply cps/kni.c]: Ah, yes. Thanks.

File: cps/kni.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (AltraMayor) [reply cps/kni.c]: In fact, if the dummy lookup fails, the blackhole entry won't be added.

**Comment by cjdoucette** [comment]
Just rebased.

**Pull request merged**